### PR TITLE
Replace older style for-loops with enhanced for-loops

### PR DESCRIPTION
**Pull request by jmark99**
Updating source code to make use of newer enhanced for-loop constructs.

### CODE REVIEW

File: core/src/main/java/org/apache/accumulo/core/client/summary/SummarizerConfiguration.java [hunk 1]
```diff
@@ -271,7 +271,7 @@ public Builder addOptions(String... keyValuePairs) {
      * @see SummarizerConfiguration#getOptions()
      */
     public Builder addOptions(Map<String,String> options) {
-      options.entrySet().forEach(e -> addOption(e.getKey(), e.getValue()));
+      options.forEach((key, value) -> addOption(key, value));
```
Comment (milleruntime) [inline core/src/main/java/org/apache/accumulo/core/client/summary/SummarizerConfiguration.java]: You can even go a step further with this one:
```suggestion
      options.forEach(this::addOption);
```

Comment (jmark99) [reply core/src/main/java/org/apache/accumulo/core/client/summary/SummarizerConfiguration.java]: done

File: test/src/main/java/org/apache/accumulo/test/SampleIT.java [hunk 2]
```diff
@@ -280,10 +279,8 @@ private String writeData(BatchWriter bw, SamplerConfiguration sc, TreeMap<Key,Va
   private int countEntries(Iterable<Entry<Key,Value>> scanner) {
 
     int count = 0;
-    Iterator<Entry<Key,Value>> iter = scanner.iterator();
 
-    while (iter.hasNext()) {
-      iter.next();
+    for (Entry<Key,Value> keyValueEntry : scanner) {
```
[the remark this answers is not in the document]
Comment (jmark99) [reply test/src/main/java/org/apache/accumulo/test/SampleIT.java]: Replaced with Iterables.size()

**Comment by jmark99** [comment]
Fixes #1495

**Comment by ctubbsii** [comment]
This was merged in 6e8787e36e5b89dbc9c5af5e2d2b35bab74b7f36

**Comment by jmark99** [comment]
@ctubbsii do you know why this page is indicating it is closed with unmerged commits even though the changes have been merged into master?

**Comment by ctubbsii** [comment]
@jmark99 The only time I've seen that is if the commit was rewritten and merged on the command-line, rather than via the web UI. If you merged it via the web UI, or didn't re-write the commit, it could just be a weird screwy thing with GitHub. I wouldn't worry about it.

**Pull request closed without merging**